lang/java/reef-utils/src/test/java/org/apache/reef/util/MultiAsyncToSyncTest.java

```diff
@@ -0,0 +1,242 @@
+
+  /**
+   * Verify calculations successfully complete when no timeout occurs.
+   */
+  @Test
+  public void testNoTimeout() {
+    LOG.log(Level.INFO, "Starting...");
+
+    // Parameters that do not force a timeout.
+    final int incrementerSleepTimeSeconds = 2;
+    final long timeoutPeriodSeconds = 4;
+    final Integer input = 1;
+
+    Integer result = 0;
+    try {
+      SynchronousApi apiObject = new SynchronousApi(incrementerSleepTimeSeconds, timeoutPeriodSeconds);
+      result = apiObject.apiCall(input);
+      apiObject.complete();
+    } catch (Exception e) {
+      LOG.log(Level.SEVERE, "Unexpected exception during test", e);
+    }
+    Assert.assertTrue("Value incremented by one", result.equals(input + 1));
+  }
+
+  /**
+   * Verify an error is returned when a timeout occurs.
+   */
+  @Test
+  public void testTimeout() {
+    LOG.log(Level.INFO, "Starting...");
+
+    // Parameters that do not force a timeout.
+    final int incrementerSleepTimeSeconds = 4;
+    final long timeoutPeriodSeconds = 2;
+    final Integer input = 1;
+
+    Integer result = 0;
+    try {
+      SynchronousApi apiObject = new SynchronousApi(incrementerSleepTimeSeconds, timeoutPeriodSeconds);
+      result = apiObject.apiCall(input);
+      apiObject.complete();
+    } catch (ExecutionException ee) {
+      if (ee.getCause() instanceof InvalidBlockedCallerIdentifierException) {
+        LOG.log(Level.INFO, "Caught expected exception during test");
+      } else {
+        LOG.log(Level.SEVERE, "Unexpected exception during test", ee);
+      }
+    } catch (Exception e) {
+      LOG.log(Level.SEVERE, "Unexpected exception during test", e);
+    }
+    Assert.assertTrue("Timeout occurred", result.equals(0));
+  }
+
+  /**
+   * Verify no interaction occurs when multiple calls are in flight.
+   */
+  @Test
+  public void testMulitpleCalls() {
+    LOG.log(Level.INFO, "Starting...");
+
+    // Parameters that do not force a timeout.
+    final int incrementerSleepTimeSeconds = 2;
+    final long timeoutPeriodSeconds = 4;
+    final Integer input = 1;
+    final ExecutorService executor = Executors.newFixedThreadPool(2);
+
+    Integer result1 = 0;
+    Integer result2 = 0;
+    try {
+      final String functionName = "apiCall";
+      final SynchronousApi apiObject = new SynchronousApi(incrementerSleepTimeSeconds, timeoutPeriodSeconds);
+      final FutureTask<Integer> task1 = new FutureTask<>(
+          new AsynchronousCaller<Integer, Integer, SynchronousApi>(apiObject, input, functionName));
+      final FutureTask<Integer> task2 = new FutureTask<>(
+          new AsynchronousCaller<Integer, Integer, SynchronousApi>(apiObject, input + 1, functionName));
+
+      // Execute API calls concurrently.
+      executor.execute(task1);
+      executor.execute(task2);
+
+      result1 = task1.get();
+      result2 = task2.get();
+
+      apiObject.complete();
+    } catch (Exception e) {
```

Comment:
Well, that's because you have to add that exception to the `throws` clause of the test method. I mean, the method should look like this:
```java
  @Test
  public void testMultipleCalls()
      throws ExecutionException, InterruptedException, NoSuchMethodException {

    // Parameters that do not force a timeout.
    final int incrementerSleepTimeSeconds = 2;
    final long timeoutPeriodSeconds = 4;
    final int input = 1;
    final ExecutorService executor = Executors.newFixedThreadPool(2);

    final String function = "apiCall";

    final SynchronousApi apiObject =
        new SynchronousApi(incrementerSleepTimeSeconds, timeoutPeriodSeconds);

    final FutureTask<Integer> task1 =
        new FutureTask<>(new MethodCallable<Integer>(apiObject, function, input));

    final FutureTask<Integer> task2 =
        new FutureTask<>(new MethodCallable<Integer>(apiObject, function, input + 1));

    // Execute API calls concurrently.
    executor.execute(task1);
    executor.execute(task2);

    final int result1 = task1.get();
    final int result2 = task2.get();

    apiObject.complete();

    Assert.assertEquals("Input must be incremented by one", input + 1, result1);
    Assert.assertEquals("Input must be incremented by one", input + 2, result2);
  }
```